lib/suse/connect/client.rb

```diff
@@ -23,6 +24,12 @@ def initialize(config)
 
       # Announces the system, activates the product on SCC and adds the service to the system
       def register!
+        file = '/etc/sysconfig/rhn/systemid'
+        if File.exist?(file)
```

Comment:
I looked through the code and I vote for moving this to the `SUSE::Connect::Cli` class (`https://github.com/SUSE/connect/blob/master/lib/suse/connect/cli.rb`) in the `execute!` method to disable the cli completely.
Otherwise we disallow also the usage of the `API` and almost sure we don't want this.

You can also use `log.error` etc.. than which removes the need to the bash sequences.

Reply:
Yeah, I can move it there.